includes/object-cache.php

```diff
@@ -566,13 +566,20 @@ public function __construct($fail_gracefully = true)
                 $this->redis_client .= sprintf(' (v%s)', Predis\Client::VERSION);
             }
 
-            if (defined('WP_REDIS_CLUSTER')) {
-                $this->redis->ping(current(array_values(WP_REDIS_CLUSTER)));
+            $isCluster = defined('WP_REDIS_CLUSTER');
+            $usesPredis = strcasecmp('predis', $client) === 0;
+
+            if ($isCluster) {
+                if (! $usesPredis) {
+                    $this->redis->ping(uniqid());
+                }
             } else {
                 $this->redis->ping();
             }
 
-            if ( ! isset( $options['replication'] ) || ! $options['replication'] ) {
+            $usesReplication = isset($options['replication']) && $options['replication'];
+
+            if (! $usesReplication) {
                 $server_info = $this->redis->info( 'SERVER' );
```

Comment:
@jack9603301: Can you please test this code with your Predis cluster and post the exception it throws, if it throws any?

Reply:
There is no judgment. I have submitted relevant corrections. Please see what I submitted. The reason why I made a judgment is that I found that the `server` instruction may have execution errors in predis mode. It is forbidden to execute the command for redis cluster.

Reply:
I will not merge this pull request, unless I understand it and see the corresponding exception/error message.

Reply:
I remember that I gave you an exception about ping command execution error, which is the same as the exception!

Reply:
I remember that I gave you an exception about ping command execution error, which is the same as the exception!